device-telemetry/app/com/microsoft/azure/iotsolutions/devicetelemetry/webservice/v1/controllers/MessagesController.java

```diff
@@ -25,47 +27,79 @@
 
     private final IMessages messages;
 
+    private static final int DEVICE_LIMIT = 1000;
+
     @Inject
     public MessagesController(final IMessages messages) {
         this.messages = messages;
     }
 
     public Result list(
-        String from,
-        String to,
-        String order,
-        Integer skip,
-        Integer limit,
-        String devices) throws
-        InvalidInputException,
-        InvalidConfigurationException,
-        TimeSeriesParseException {
+            String from,
+            String to,
+            String order,
+            Integer skip,
+            Integer limit,
+            String devices) throws
+            InvalidInputException,
+            InvalidConfigurationException,
+            TimeSeriesParseException {
 
         DateTime fromDate = DateHelper.parseDate(from);
         DateTime toDate = DateHelper.parseDate(to);
 
-        if (order == null) order = "asc";
-        if (skip == null) skip = 0;
-        if (limit == null) limit = 1000;
-
-        // TODO: move this logic to the storage engine, depending on the
-        // storage type the limit will be different. 200 is CosmosDb
-        // limit for the IN clause.
         String[] deviceIds = new String[0];
         if (devices != null) {
             deviceIds = devices.split(",");
         }
-        if (deviceIds.length > 200) {
+
+        return this.listMessagesHelper(fromDate, toDate, order, skip, limit, deviceIds);
+    }
+
+    public Result post() throws InvalidConfigurationException, InvalidInputException, TimeSeriesParseException {
```

Comment:
Will this also need an `@Authorize` tag?

Reply:
yep, was waiting for other changes to go in. updated